src/navier.f90

```diff
@@ -1485,33 +1447,12 @@ subroutine pipe_bulk_phi(phi,ux,ep,is,phib_constant)
 
     !--------------------------- Bulk Temperature ---------------------------
     !                  with corrected streamwise velocity
-    qm=zero
-    qv=zero
-    ncount=zero
-    do k=1,xsize(3)
-        zm=dz*real(xstart(3)-1+k-1,mytype)-zc
-        do j=1,xsize(2)
-            if (istret.eq.0) ym=real(j+xstart(2)-1-1,mytype)*dy-yc
-            if (istret.ne.0) ym=yp(j+xstart(2)-1)-yc
-            r=sqrt(ym*ym+zm*zm)
-            do i=1,xsize(1)
-                if (r.le.rai.and.ep(i,j,k).eq.0) then
-                    qm=qm+ux(i,j,k)*phi(i,j,k)
-                    qv=qv+ux(i,j,k)*ux(i,j,k)
-                    ncount=ncount+one
-                endif
-            enddo
-        enddo
-    enddo
-    call MPI_ALLREDUCE(qm,qmm,1,real_type,MPI_SUM,MPI_COMM_WORLD,code)
-    call MPI_ALLREDUCE(qv,qvm,1,real_type,MPI_SUM,MPI_COMM_WORLD,code)
-    call MPI_ALLREDUCE(ncount,ncountt,1,real_type,MPI_SUM,MPI_COMM_WORLD,code)
-    qmm=qmm/ncountt
-    qvm=qvm/ncountt
+    call pipe_volume_avg(ux*phi,qm,ep)
+    call pipe_volume_avg(ux*ux ,qv,ep)
```

Comment:
A small improvement could be to compute `ncount` only one time, in the subroutine `pipe_bulk`. And to pass it downward as a argument ? This would avoid repeated MPI_ALL reduction on `ncount`.